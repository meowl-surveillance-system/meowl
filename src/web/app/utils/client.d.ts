import cassandra from 'cassandra-driver';
export declare const client: cassandra.Client;
