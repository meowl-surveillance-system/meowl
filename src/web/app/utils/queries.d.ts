export declare const INSERT_USERSID = "INSERT INTO users_id (user_id, email, username, password, sid) VALUES (?, ?, ?, ?, ?)";
export declare const INSERT_USERSNAME = "INSERT INTO users_name (user_id, email, username, password, sid) VALUES (?, ?, ?, ?, ?)";
export declare const SELECT_USERSID_USERID_PASSWORD = "SELECT user_id, password FROM users_name WHERE username = ?";
export declare const UPDATE_USERSID_SID = "UPDATE users_id SET sid = ? WHERE user_id = ?";
export declare const UPDATE_USERSNAME_SID = "UPDATE users_name SET sid = ? WHERE username = ?";
export declare const SELECT_USERSNAME_SID = "SELECT sid FROM users_id WHERE user_id = ?";
export declare const SELECT_SID_SESSION = "SELECT * FROM sessions WHERE sid = ?";
export declare const SELECT_USERSNAME_USERID = "SELECT user_id FROM users_name WHERE username = ?";
export declare const SELECT_CAMERAID_USERID = "SELECT camera_id FROM user_cameras WHERE user_id = ?";
export declare const SELECT_USERID_CAMERAID = "SELECT user_id FROM camera_users WHERE camera_id = ?";
export declare const INSERT_CAMERAID_USERID = "INSERT INTO user_cameras (user_id, camera_id) VALUES (?, ?)";
export declare const INSERT_USERID_CAMERAID = "INSERT INTO camera_users (user_id, camera_id) VALUES (?, ?)";
export declare const INSERT_CAMERAID_STREAMID = "INSERT INTO camera_streams (camera_id, stream_id, stream_date) VALUES(?, ?, ?)";
export declare const SELECT_CAMERAID_STREAMID = "SELECT stream_id FROM camera_streams WHERE camera_id = ?";
export declare const SELECT_STREAMID_METADATA = "SELECT stream_id FROM metadata WHERE stream_id = ?";
export declare const SELECT_CAMERAID_STREAMID_SINGLE = "SELECT stream_id FROM camera_streams WHERE camera_id = ? LIMIT 1";
export declare const INSERT_CAMERAID_LIVE = "INSERT INTO live_cameras (camera_id, live) VALUES (?, ?)";
export declare const SELECT_CAMERAID = "SELECT DISTINCT camera_id FROM camera_streams";
export declare const SELECT_LIVE_CAMERAID = "SELECT live FROM live_cameras WHERE camera_id = ?";
