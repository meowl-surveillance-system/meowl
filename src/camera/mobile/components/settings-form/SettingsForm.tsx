import React from 'react';
import { Modal, Button } from 'react-native';
import { Text, Input, ButtonGroup } from 'react-native-elements';

class SettingsForm extends React.Component<SettingsFormProps> {
  constructor(props: SettingsFormProps) {
    super(props);

  }

  updateProps(props: object) {
    this.props.updateProps(props);
  }

  render() {
    const fpsBtns: string[] = ['15', '20', '24', '30'];
    const fpsIndexMap: { [fps: number]: number } = {
      15: 0,
      20: 1,
      24: 2,
      30: 3,
    }

    return (
      <Modal
        animationType="slide"
        transparent={false}
        visible={this.props.modalVisible}
      >
        <Text style={{ fontSize: 22 }}>
          Stream Link:
          </Text>
        <Input
          onChangeText={(outputLink) => this.updateProps({ outputLink: outputLink })}
          value={this.props.outputLink}
          placeholder="rtmp://"
        />
        <Text style={{ fontSize: 22 }}>
          Video Bit Rate:
        </Text>
        <Input
          onChangeText={(videoBitRate) => this.updateProps({ videoBitRate: parseInt(videoBitRate) })}
          value={this.props.videoBitRate.toString()}
          placeholder="8000000 (Recommended)"
        /><Text style={{ fontSize: 22 }}>
          Audio Bit Rate:
      </Text>
        <Input
          onChangeText={(audioBitRate) => this.updateProps({ audioBitRate: parseInt(audioBitRate) })}
          value={this.props.audioBitRate.toString()}
          placeholder="128000 (Recommended)"
        />
        <Text style={{ fontSize: 22 }}>
          Frames Per Second
          </Text>
        <ButtonGroup
          onPress={(selectedIndex) => this.updateProps({ fps: parseInt(fpsBtns[selectedIndex]) })}
          selectedIndex={fpsIndexMap[this.props.fps]}
          buttons={fpsBtns}
          containerStyle={{ height: 100 }}
        />
        <Button
          onPress={() => {
            this.updateProps({ modalVisible: false });
          }}
          title="Done"
        />
      </Modal>
    );
  }
}

export default SettingsForm;