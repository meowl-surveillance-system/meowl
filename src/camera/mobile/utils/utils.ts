import AsyncStorage from '@react-native-community/async-storage';
/**
 * Makes a request to log out from the server
 * @param serverUrl - Server URL that is hosting the requests
 * @param logoutEndpoint - Endpoint where the logout method is called
 * @returns the response to log out from the server
 */
export async function logout(
  serverUrl: string,
  logoutEndpoint = '/auth/logout',
) {
  const logoutUrl: string = serverUrl + logoutEndpoint;
  console.log(logoutUrl);
  try {
    const logoutResponse = await fetch(logoutUrl, {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
        credentials: 'include',
      },
    });
    return logoutResponse;
  } catch (err) {
    return null;
  }
}

/**
 * Makes a request to log into the server
 * @param credentials - Credentials required to login (e.g. username/password)
 * @param serverUrl - Server URL that is hosting the requests
 * @param loginEndpoint - Endpoint where the login method is called
 * @returns the response to log into the server
 */
export async function login(
  credentials: object,
  serverUrl: string,
  loginEndpoint = '/auth/login',
) {
  const loginUrl = serverUrl + loginEndpoint;
  try {
    const loginResponse = await fetch(loginUrl, {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
        credentials: 'include',
      },
      body: JSON.stringify(credentials),
    });
    return loginResponse;
  } catch (err) {
    return null;
  }
}

/**
 * Get item from AsyncStorage if found else return value from args
 * 
 * @param key - Key to retrieve value from AsyncStorage
 * @param value - value to be used if value not found in AsyncStorage
 * @returns value
 */
export async function getItemFromStore(
  key: string,
  value: string = ''
): Promise<string> {
  const valueFromStore: string | null = await AsyncStorage.getItem(key);
  if (valueFromStore) {
    return valueFromStore;
  }
  return value;
}