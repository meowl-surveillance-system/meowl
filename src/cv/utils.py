import json
import requests
import urllib

def get_auth_RTMP_url(
  rtmpUrl,
  loginUrl,
  rtmpRequestUrl,
  username,
  password):
  """Authenticate to loginUrl, append authorization parameters to rtmpUrl"""
  print(username, password)
  body = {"username": username, "password": password}
  loginResponse = requests.post(loginUrl, data=body)
  if not loginResponse.ok:
    raise Exception(f"Auth server response not ok : {loginResponse.text}")
  connectsidCookie = {
    "connect.sid": loginResponse.cookies["connect.sid"]}
  rtmpRequestResponse = requests.post(
    rtmpRequestUrl, cookies=connectsidCookie)
  if not rtmpRequestResponse.ok:
    raise Exception(f"Auth server response not ok : {rtmpRequestResponse.text}")
  rtmpCreds = json.loads(rtmpRequestResponse.text)
  return rtmpUrl + "?" + urllib.parse.urlencode(rtmpCreds)