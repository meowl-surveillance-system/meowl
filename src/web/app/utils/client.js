"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const cassandra_driver_1 = __importDefault(require("cassandra-driver"));
exports.client = new cassandra_driver_1.default.Client({
    contactPoints: ['127.0.0.1'],
    localDataCenter: 'datacenter1',
    keyspace: 'streams',
});
//# sourceMappingURL=client.js.map