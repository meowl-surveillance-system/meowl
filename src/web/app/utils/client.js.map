{"version":3,"file":"client.js","sourceRoot":"","sources":["../../../app/utils/client.ts"],"names":[],"mappings":";;;;;AAAA,wEAAyC;AAE5B,QAAA,MAAM,GAAG,IAAI,0BAAS,CAAC,MAAM,CAAC;IACzC,aAAa,EAAE,CAAC,WAAW,CAAC;IAC5B,eAAe,EAAE,aAAa;IAC9B,QAAQ,EAAE,SAAS;CACpB,CAAC,CAAC"}