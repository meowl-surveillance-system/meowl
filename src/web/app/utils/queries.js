"use strict";
// Query names take the following form: <ACTION>_<TABLE NAME>_<COLUMNS YOU WANT
// TO DO THE ACTION ON>
Object.defineProperty(exports, "__esModule", { value: true });
exports.INSERT_USERSID = 'INSERT INTO users_id (user_id, email, username, password, sid) VALUES (?, ?, ?, ?, ?)';
exports.INSERT_USERSNAME = 'INSERT INTO users_name (user_id, email, username, password, sid) VALUES (?, ?, ?, ?, ?)';
exports.SELECT_USERSID_USERID_PASSWORD = 'SELECT user_id, password FROM users_name WHERE username = ?';
exports.UPDATE_USERSID_SID = 'UPDATE users_id SET sid = ? WHERE user_id = ?';
exports.UPDATE_USERSNAME_SID = 'UPDATE users_name SET sid = ? WHERE username = ?';
exports.SELECT_USERSNAME_SID = 'SELECT sid FROM users_id WHERE user_id = ?';
exports.SELECT_SID_SESSION = 'SELECT * FROM sessions WHERE sid = ?';
exports.SELECT_USERSNAME_USERID = 'SELECT user_id FROM users_name WHERE username = ?';
exports.SELECT_CAMERAID_USERID = 'SELECT camera_id FROM user_cameras WHERE user_id = ?';
exports.SELECT_USERID_CAMERAID = 'SELECT user_id FROM camera_users WHERE camera_id = ?';
exports.INSERT_CAMERAID_USERID = 'INSERT INTO user_cameras (user_id, camera_id) VALUES (?, ?)';
exports.INSERT_USERID_CAMERAID = 'INSERT INTO camera_users (user_id, camera_id) VALUES (?, ?)';
exports.INSERT_CAMERAID_STREAMID = 'INSERT INTO camera_streams (camera_id, stream_id, stream_date) VALUES(?, ?, ?)';
exports.SELECT_CAMERAID_STREAMID = 'SELECT stream_id FROM camera_streams WHERE camera_id = ?';
exports.SELECT_STREAMID_METADATA = 'SELECT stream_id FROM metadata WHERE stream_id = ?';
exports.SELECT_CAMERAID_STREAMID_SINGLE = 'SELECT stream_id FROM camera_streams WHERE camera_id = ? LIMIT 1';
exports.INSERT_CAMERAID_LIVE = 'INSERT INTO live_cameras (camera_id, live) VALUES (?, ?)';
exports.SELECT_CAMERAID = 'SELECT DISTINCT camera_id FROM camera_streams';
exports.SELECT_LIVE_CAMERAID = 'SELECT live FROM live_cameras WHERE camera_id = ?';
//# sourceMappingURL=queries.js.map