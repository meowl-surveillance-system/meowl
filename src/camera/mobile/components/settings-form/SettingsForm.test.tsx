import 'react-native';
import React from 'react';
import SettingsForm from './SettingsForm';
import * as utils from '../../utils/utils';

import renderer from 'react-test-renderer';

describe('SettingsForm Component', () => {
  let settingsFormInstance: SettingsForm;
  let updatePropsMock: jest.Mock;

  beforeEach(() => {
    updatePropsMock = jest.fn();
    settingsFormInstance = new SettingsForm({
      audioBitRate: 1234,
      fps: 1234,
      requestServerUrl: '',
      rtmpServerUrl: '',
      settingsFormVisible: false,
      updateProps: updatePropsMock,
      videoBitRate: 1234,
    });
  });

  it('renders correctly', () => {
    const container = renderer.create(
      <SettingsForm
        audioBitRate={1234}
        fps={1234}
        requestServerUrl=''
        rtmpServerUrl=''
        settingsFormVisible={false}
        updateProps={(obj) => { }}
        videoBitRate={1234}
      />
    ).toJSON();
    expect(container).toMatchSnapshot();
  });

  it('should set isLoggedIn to false on logout', async () => {
    const logOutSpy: jest.SpyInstance = jest.spyOn(utils, 'logout');
    await settingsFormInstance.handleLogout();
    expect(logOutSpy).toHaveBeenCalled();
    expect(updatePropsMock).toHaveBeenCalledWith({ isLoggedIn: false });
  });

  it('should call props.updateProps on updates', async () => {
    settingsFormInstance.updateProps({ rtmpServerUrl: 'testing1234567' });
    expect(updatePropsMock).toHaveBeenLastCalledWith({ rtmpServerUrl: 'testing1234567' });
  });

});