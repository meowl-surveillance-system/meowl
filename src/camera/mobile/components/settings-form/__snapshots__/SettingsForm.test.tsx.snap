// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  transparent={false}
  visible={false}
>
  <Text
    style={
      Object {
        "fontSize": 22,
      }
    }
    theme={
      Object {
        "colors": Object {
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff190c",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "platform": Object {
            "android": Object {
              "error": "#f44336",
              "primary": "#2196f3",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "ios": Object {
              "error": "#ff3b30",
              "primary": "#007aff",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
          },
          "primary": "#2089dc",
          "searchBg": "#303337",
          "secondary": "#8F0CE8",
          "success": "#52c41a",
          "warning": "#faad14",
        },
      }
    }
  >
    Stream Link:
  </Text>
  <View
    style={
      Object {
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        editable={true}
        onChangeText={[Function]}
        placeholder="rtmp://"
        rejectResponderTermination={true}
        style={
          Object {
            "alignSelf": "center",
            "color": "black",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="RNE__Input__text-input"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <Text
    style={
      Object {
        "fontSize": 22,
      }
    }
    theme={
      Object {
        "colors": Object {
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff190c",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "platform": Object {
            "android": Object {
              "error": "#f44336",
              "primary": "#2196f3",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "ios": Object {
              "error": "#ff3b30",
              "primary": "#007aff",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
          },
          "primary": "#2089dc",
          "searchBg": "#303337",
          "secondary": "#8F0CE8",
          "success": "#52c41a",
          "warning": "#faad14",
        },
      }
    }
  >
    Video Bit Rate:
  </Text>
  <View
    style={
      Object {
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        editable={true}
        onChangeText={[Function]}
        placeholder="8000000 (Recommended)"
        rejectResponderTermination={true}
        style={
          Object {
            "alignSelf": "center",
            "color": "black",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="RNE__Input__text-input"
        underlineColorAndroid="transparent"
        value="1234"
      />
    </View>
  </View>
  <Text
    style={
      Object {
        "fontSize": 22,
      }
    }
    theme={
      Object {
        "colors": Object {
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff190c",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "platform": Object {
            "android": Object {
              "error": "#f44336",
              "primary": "#2196f3",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "ios": Object {
              "error": "#ff3b30",
              "primary": "#007aff",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
          },
          "primary": "#2089dc",
          "searchBg": "#303337",
          "secondary": "#8F0CE8",
          "success": "#52c41a",
          "warning": "#faad14",
        },
      }
    }
  >
    Audio Bit Rate:
  </Text>
  <View
    style={
      Object {
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        editable={true}
        onChangeText={[Function]}
        placeholder="128000 (Recommended)"
        rejectResponderTermination={true}
        style={
          Object {
            "alignSelf": "center",
            "color": "black",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="RNE__Input__text-input"
        underlineColorAndroid="transparent"
        value="1234"
      />
    </View>
  </View>
  <Text
    style={
      Object {
        "fontSize": 22,
      }
    }
    theme={
      Object {
        "colors": Object {
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff190c",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "platform": Object {
            "android": Object {
              "error": "#f44336",
              "primary": "#2196f3",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "ios": Object {
              "error": "#ff3b30",
              "primary": "#007aff",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
          },
          "primary": "#2089dc",
          "searchBg": "#303337",
          "secondary": "#8F0CE8",
          "success": "#52c41a",
          "warning": "#faad14",
        },
      }
    }
  >
    Frames Per Second
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#e3e3e3",
        "borderRadius": 3,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 100,
        "marginBottom": 5,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 5,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomLeftRadius": 3,
          "borderRightColor": "#bdc6cf",
          "borderRightWidth": 0,
          "borderTopLeftRadius": 3,
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
          }
        }
        testID="buttonGroupItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#5e6977",
                "fontSize": 16.25,
                "fontWeight": "500",
              }
            }
            testID="buttonGroupItemText"
            theme={
              Object {
                "colors": Object {
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "primary": "#2196f3",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "primary": "#007aff",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              }
            }
          >
            15
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderLeftColor": "#bdc6cf",
          "borderLeftWidth": 1,
          "borderRightColor": "#bdc6cf",
          "borderRightWidth": 1,
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
          }
        }
        testID="buttonGroupItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#5e6977",
                "fontSize": 16.25,
                "fontWeight": "500",
              }
            }
            testID="buttonGroupItemText"
            theme={
              Object {
                "colors": Object {
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "primary": "#2196f3",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "primary": "#007aff",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              }
            }
          >
            20
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderRightColor": "#bdc6cf",
          "borderRightWidth": 1,
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
          }
        }
        testID="buttonGroupItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#5e6977",
                "fontSize": 16.25,
                "fontWeight": "500",
              }
            }
            testID="buttonGroupItemText"
            theme={
              Object {
                "colors": Object {
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "primary": "#2196f3",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "primary": "#007aff",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              }
            }
          >
            24
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderBottomRightRadius": 3,
          "borderTopRightRadius": 3,
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "opacity": 1,
          }
        }
        testID="buttonGroupItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#5e6977",
                "fontSize": 16.25,
                "fontWeight": "500",
              }
            }
            testID="buttonGroupItemText"
            theme={
              Object {
                "colors": Object {
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "primary": "#2196f3",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "primary": "#007aff",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              }
            }
          >
            30
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Logout
      </Text>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Done
      </Text>
    </View>
  </View>
</Modal>
`;
